### ISSUE TITLE
Accept functional interface as loss parameter

### ISSUE THREAD

**Issue by floffy-f** [iutzeler/skwdro#164]
## Current state

Only `nn.Module`s are accepted in the main interfaces accepting `(loss, transformation)` pairs.
This need is made explicit in the documentation.

## Goal

Accept `nn.functional` functions as parameters, marked as `Callable`s.
The forward pass should account for this.

## Plan

- [x] Change `WrappedPrimalLoss` to make it adapt to the two cases `nn.Module` vs `nn.functional`.
- [x] Update the typing:
    - [x] `DualLoss`es
    - [x] `robustify`/`dualize_primal_loss`
- [x] Remove mentions of this limitation in the doc
    - [x]  see warning block in the pytorch interface tutorial
